### PR TITLE
Updating icon imports to TS version in component-library folder

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## Explanation
A part of the TypeScript migration work. This PR updates 
- All previously deprecated imports to the new TS version of the component and enums e.g. `Icon`, `IconName`, `IconSize`
- Updates all propTypes that used `Icon.propTypes` to `object`
- Update all docs that say ICON_NAMES object to `IconName` enum

The enums being used in this PR are here
`IconSize` https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/component-library/icon/icon.types.ts#L4-L11
`IconName` https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/component-library/icon/icon.types.ts#L22-L150

* Part of #18326 
* See: https://github.com/MetaMask/metamask-extension/pull/18431
* https://consensys.slack.com/archives/G1L7H42BT/p1680299226290989

## Screenshots/Screencaps

No change in icon display. Icon docs have been updated accordingly screencasts show no change in stories.

### Before

https://user-images.githubusercontent.com/8112138/229937890-2af9799c-39d2-4fda-8631-489b1ed283cb.mov

### After

https://user-images.githubusercontent.com/8112138/229937647-f5a3d5c2-abeb-4358-95e0-729a90853294.mov

## Manual Testing Steps

- Go to the storybook build of this PR 
- Go through stories that use the Icon component
- Check icon display and controls

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/component-library/avatar-favicon/avatar-favicon.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ AvatarFavicon.propTypes = {
   /**
    * Props for the fallback icon. All Icon props can be used
    */
-  fallbackIconProps: PropTypes.shape(Icon.PropTypes),
+  fallbackIconProps: PropTypes.object,
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-favicon/avatar-favicon.js:68]: Not a huge fan of this and the other options I thought of was:
- Import the deprecated `Icon` component as `IconDeprecated` and use it's propTypes but I'm conscious of those propTypes becoming out of date. 
We can't use the `IconProps` interface until this component has been migrated to TS. So maybe we just go through a short period of unsafe types. I don't think this prop will be regularly used it's more of an escape hatch.

Do you have any suggestions @garrettbear @brad-decker?

File: ui/components/component-library/avatar-icon/avatar-icon.js [hunk 2]
```diff
@@ -28,6 +28,7 @@ export const AvatarIcon = ({
   iconName,
   ...props
 }) => {
+  console.log('IconName', IconName);
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-icon/avatar-icon.js]: Remove console.log

File: ui/components/component-library/avatar-icon/avatar-icon.stories.js [hunk 3]
```diff
@@ -107,80 +106,82 @@ export const SizeStory = (args) => (
...
       {...args}
     />
     <AvatarIcon
       color={IconColor.errorDefault}
       backgroundColor={BackgroundColor.errorMuted}
-      iconName={ICON_NAMES.DANGER}
+      iconName={IconName.Danger}
       {...args}
     />
   </Box>
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-icon/avatar-icon.stories.js:187]: ## After 

https://user-images.githubusercontent.com/8112138/229933318-6e9e0824-b7ba-4ec0-8a04-8eafd7c91df9.mov

File: ui/components/component-library/banner-alert/banner-alert.stories.js [hunk 4]
```diff
@@ -165,7 +164,7 @@ ActionButton.args = {
   actionButtonLabel: 'Action',
   actionButtonOnClick: () => console.log('ButtonLink actionButtonOnClick demo'),
   actionButtonProps: {
-    endIconName: ICON_NAMES.ARROW_2_RIGHT,
+    endIconName: IconName.Arrow2Right,
   },
   children:
     'Use actionButtonLabel for action text, actionButtonOnClick for the onClick handler, and actionButtonProps to pass any ButtonLink prop types such as iconName',
```
Comment (georgewrmarshall) [inline ui/components/component-library/banner-alert/banner-alert.stories.js:170]: ## After

https://user-images.githubusercontent.com/8112138/229933479-c05919c6-49d1-468e-b52a-c7e2d9e1c59d.mov

File: ui/components/component-library/avatar-icon/README.mdx [hunk 5]
```diff
@@ -55,22 +55,22 @@ import { AvatarIcon, ICON_NAMES } from '../ui/component-library';
...
 ### Icon Name<span style={{ color: 'red' }}>\*</span>
 
-Use the required `iconName` prop with `ICON_NAMES` object from `./ui/components/component-library` to select icon
+Use the required `iconName` prop with `IconName` enum from `./ui/components/component-library` to select icon
 
 Use the [IconSearch](/story/components-componentlibrary-icon--default-story) story to find the icon you want to use.
 
 <Canvas>
-  <Story id="components-componentlibrary-avataricon--icon-name" />
+  <Story id="components-componentlibrary-avataricon--icon-name-story" />
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-icon/README.mdx:63]: ## After

Screenshot 2023-04-04 at 3 08 51 PM

File: ui/components/component-library/avatar-icon/avatar-icon.js [hunk 6]
```diff
@@ -27,39 +27,37 @@ export const AvatarIcon = ({
   iconProps,
   iconName,
   ...props
-}) => {
-  return (
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-icon/avatar-icon.js:31]: Just removing unneeded return here. Sorry makes it harder to review.

File: ui/components/component-library/banner-base/banner-base.js [hunk 7]
```diff
@@ -72,7 +72,7 @@ export const BannerBase = ({
         <ButtonIcon
           className="mm-banner-base__close-button"
           marginLeft="auto"
-          iconName={ICON_NAMES.CLOSE}
+          iconName={IconName.Close}
```
Comment (georgewrmarshall) [inline ui/components/component-library/banner-base/banner-base.js:75]: Screenshot 2023-04-04 at 3 11 32 PM

File: ui/components/component-library/button-icon/README.mdx [hunk 8]
```diff
@@ -17,19 +17,19 @@ The `ButtonIcon` accepts all props below as well as all [Box](/docs/components-u
...
 ### Icon Name<span style={{ color: 'red' }}>\*</span>
 
-Use the required `iconName` prop with `ICON_NAMES` object from `./ui/components/component-library/icon` to select icon.
+Use the required `iconName` prop with `IconName` enum from `./ui/components/component-library` to select icon.
 
 Use the [IconSearch](/story/components-componentlibrary-icon--default-story) story to find the icon you want to use.
 
 <Canvas>
-  <Story id="components-componentlibrary-buttonicon--icon-name" />
+  <Story id="components-componentlibrary-buttonicon--icon-name-story" />
```
Comment (georgewrmarshall) [inline ui/components/component-library/button-icon/README.mdx:25]: Screenshot 2023-04-04 at 3 13 57 PM

File: ui/components/component-library/avatar-icon/avatar-icon.js (hunk 2 above)
Comment (georgewrmarshall) [reply ui/components/component-library/avatar-icon/avatar-icon.js]: removed

File: ui/components/component-library/help-text/README.mdx [hunk 9]
```diff
@@ -28,17 +28,16 @@ The `HelpText` accepts all props below as well as all [Box](/docs/components-ui-
...
 <HelpText>Plain text</HelpText> // renders as <p>Plain text</p>
 <HelpText>
   <span>Text and icon</span>
   <Icon
     marginLeft={1}
     color={IconColor.iconAlternative}
-    name={ICON_NAMES.WARNING}
-    size={Size.inherit}
+    name={IconName.Warning}
+    size={IconSizeSize.Inherit}
```
Comment (georgewrmarshall) [inline ui/components/component-library/help-text/README.mdx:40]: Update to `IconSize.Inherit`

**Review by DDDDDanica — APPROVED** [review]

**Comment by DDDDDanica** [comment]
LGTM !! 👍

File: ui/components/component-library/help-text/README.mdx (hunk 9 above)
Comment (georgewrmarshall) [reply ui/components/component-library/help-text/README.mdx]: done

**Review by garrettbear — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**